website/blog/2024-11-19-ai-lab-first-app.md

```diff
@@ -0,0 +1,220 @@
+
+Let's start a new playground (from the `Models > Playgrounds` menu), and send our first prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image"
+```
+
+The model should reply with some list of pages, in a human-readable form (see the screenshot below, for the response we received).
+
+![a first prompt with human-readable output](./img/ai-lab-first-app/04-ai-lab-demo-prompt-1.png)
+
+The problem is that the response is in human-readable form, but we don't want the API to return this response as is. We want to have the name and the url of the pages, and send them to the website, so the website can display these pages with its preferred format. 
+
+For this, we can try to ask the model to reply with a structured response, with the following prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image" as JSON output as an array of objects with 2 fields name and url
+```
+
+This time, we received a response in JSON format, which is more suitable for our needs.
+
+![a prompt with structured output](./img/ai-lab-first-app/05-ai-lab-demo-prompt-json.png)
+
+We don't expect the user to ask such a precise question, and we would prefer to send to the model the exact question of the user, without modifying it in real time. To achieve this, chat models provide a system prompt feature. The system prompt can be defined at the beginning of the chat session.
+
+Podman AI Lab supports this feature, let's restart a Playground session with the following system prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to the request as JSON output as an array of objects with 2 fields name and url
+```
+
+Then, send the prompt `build an image`, to simulate a realistic search input of a user.
+
+We can see in the screenshot below that the model still returns a response suitable for our use case.
+
+![a session with a system prompt](./img/ai-lab-first-app/06-ai-lab-demo-system-prompt.png)
+
+> Please note that this section was not a course on writing the best prompt, I'm sure you will find much more efficient prompts for this purpose. The purpose of this section is to demonstrate how you can iterate with Podman AI Lab to refine the prompts you want to use for your application.
+
+## Testing a recipe
+
+Now that we have a suitable prompt to use for our application, it is time to start our application itself. 
+
+Many developers prefer to have a working example of application to start with, and Podman AI Lab provides such examples with a catalog of recipes, visible in the page `AI Apps > Recipe Catalog`.
+
+Let's select the Chatbot recipe (`More details` link on the Chatbot card), and start it with the Mistral model (by pressing the `Start` button and filling the form).
```

Comment:
Any reason not to use a Granite model here ?

Reply:
I tested with granite but it struggles responding in JSON format